engine/lib/concrete_ident/concrete_ident.ml

```diff
@@ -346,7 +346,10 @@ module View = struct
         path
     in
     let sep = "__" in
-    let subst = String.substr_replace_all ~pattern:sep ~with_:(sep ^ "_") in
+    let subst =
```

Comment:
Are those changes for the new core library?

Reply:
This is just to get consistent naming in Coq, since I cannot have the same name twice. However, I still have some cases, where a constructor is refered to as `<name>` instead of `t_<name>`.

Reply:
I don't get it, can you provide a small example and the translation we want?

Reply:
The naming is not consistent with how it is used in Coq backend, furthermore I get collisions of names. This disambiguates the names and makes it (more) consistent across different uses. I could move this into the Coq backend, but then the naming logic would be in many different functions.

Reply:
The line `>> String.substr_replace_all ~pattern:"___" ~with_:""` was to fix issues with `___self` occuring in require statements. However, this was a fix for annotated core lib. The rest is issues with naming for the coq backend.

Reply:
Mmmh. That's not clear to me.
Let's pull those changes out of this PR, make an issue describing the problem with an example of a collision, and then push those changes to a separate PR.
That will help I think.